### PR TITLE
feat: add EarnOpportunityTypeBase

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Quality of life PR improving `EarnOpportunityType` to be a union of `StakingEarnOpportunityType | LpEarnOpportunityType`, now that all DeFi earn opportunities use the DeFi abstraction and are either LP or staking earn opportunities

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

None

## Testing

None

### Engineering

- CI is happy

### Operations

- None

### CODE REVIEW

File: src/state/slices/opportunitiesSlice/types.ts [hunk 1]
```diff
@@ -142,7 +142,7 @@ export type GetOpportunityUserStakingDataOutput = {
 export type GetOpportunityIdsOutput = OpportunityId[]
 
 // TODO: This is not FDA-approved and should stop being consumed to make things a lot tidier without the added cholesterol
```
Comment (0xApotheosis) [inline src/state/slices/opportunitiesSlice/types.ts:144]: We can probably remove this (wonderful) comment now, yeah?

File: src/state/slices/opportunitiesSlice/types.ts [hunk 2]
```diff
@@ -142,7 +142,7 @@ export type GetOpportunityUserStakingDataOutput = {
...
-export type EarnOpportunityType = {
+export type EarnOpportunityTypeBase = {
```
Comment (0xApotheosis) [inline src/state/slices/opportunitiesSlice/types.ts:145]: We can now remove the `export`.

**Review by 0xApotheosis — APPROVED** [review]

File: src/state/slices/opportunitiesSlice/types.ts (hunk 1 above)
Comment (gomesalexandre) [reply src/state/slices/opportunitiesSlice/types.ts:144]: Unfortunately not 😢 we're still heavily relying on "earn opportunities" as a legacy of previous implementations.

The rationale was we should now be able to derive everything at component-level, from the opportunitiesSlice raw types.

File: src/state/slices/opportunitiesSlice/types.ts (hunk 2 above)
Comment (gomesalexandre) [reply src/state/slices/opportunitiesSlice/types.ts]: 9a92da2c90f797da0e7ba487bdbbf155c573e438

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**